sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -369,6 +387,10 @@ class AstBuilder extends DataTypeAstBuilder
         if Option(c.beginLabel()).isDefined &&
           c.beginLabel().multipartIdentifier().getText.toLowerCase(Locale.ROOT).equals(label)
         => true
+      case c: ForStatementContext
+        if Option(c.beginLabel()).isDefined &&
```

Comment:
you can express it as

```
if Option(c.beginLabel).map { b =>
  b.multipartIdentifier().getText...)
  }.getOrElse(false)
```

this way you don't have to repeat the name of the field.